turbine/src/sigverify_shreds.rs

```diff
@@ -175,18 +218,64 @@ fn run_shred_sigverify<const K: usize>(
     Ok(())
 }
 
+#[must_use]
+fn verify_retransmitter_signature(
+    shred: &[u8],
+    root_bank: &Bank,
+    working_bank: &Bank,
+    cluster_info: &ClusterInfo,
+    leader_schedule_cache: &LeaderScheduleCache,
+    cluster_nodes_cache: &ClusterNodesCache<RetransmitStage>,
+    stats: &ShredSigVerifyStats,
+) -> bool {
+    let signature = match shred::layout::get_retransmiter_signature(shred) {
+        Ok(signature) => signature,
+        // If the shred is not of resigned variant,
+        // then there is nothing to verify.
+        Err(shred::Error::InvalidShredVariant) => return true,
+        Err(_) => return false,
+    };
+    let Some(merkle_root) = shred::layout::get_merkle_root(shred) else {
+        return false;
+    };
+    let Some(shred) = shred::layout::get_shred_id(shred) else {
+        return false;
+    };
+    let Some(leader) = leader_schedule_cache.slot_leader_at(shred.slot(), Some(working_bank))
+    else {
+        stats
+            .num_unknown_slot_leader
+            .fetch_add(1, Ordering::Relaxed);
+        return false;
+    };
+    let cluster_nodes =
+        cluster_nodes_cache.get(shred.slot(), root_bank, working_bank, cluster_info);
+    let data_plane_fanout = cluster_nodes::get_data_plane_fanout(shred.slot(), root_bank);
```

Comment:
I'm not smart enough to think through whether root_bank or working_bank is the right thing to use here. I'm assuming you know better and/or have tested epoch boundary crossing :)

Reply:
I think we should only use the `root_bank`.

`working_bank` was historically added because at epoch boundary we were not able to obtain necessary epoch stakes for shreds received for the next epoch but that should have been fixed by https://github.com/solana-labs/solana/pull/32743.

Below PRs have some context:
https://github.com/solana-labs/solana/pull/32743
https://github.com/solana-labs/solana/pull/18971#discussion_r681930509